File: tatala-client-csharp/src/Example/Tatala/proxy/ChatRoomClientDefaultProxy.cs
using System;
using Example.tatala.client;
using QiLeYuan.Tatala.proxy;
using QiLeYuan.Tatala.socket.to;

namespace Example.tatala.proxy {
    public class ChatRoomClientDefaultProxy : DefaultProxy {

        public override Object execute(TransferObject baseto) {
            StandardTransferObject to = (StandardTransferObject)baseto;
            String calleeMethod = to.getCalleeMethod();
            if (calleeMethod.Equals("receiveMessage")) {
                String message = to.getString("message");
                ChatRoomClient.getInstance().receiveMessage(message);
            }

            return null;
        }
    }
}

File: tatala-java-example/src/com/qileyuan/tatala/example/proxy/ChatRoomClientDefaultProxy.java
package com.qileyuan.tatala.example.proxy;

import com.qileyuan.tatala.example.client.ChatRoomClient;
import com.qileyuan.tatala.proxy.DefaultProxy;
import com.qileyuan.tatala.socket.to.TransferObject;

public class ChatRoomClientDefaultProxy extends DefaultProxy {
	
	public Object execute(TransferObject to){
		String calleeMethod = to.getCalleeMethod();
		if(calleeMethod.equals("receiveMessage")){
			String message = to.getString();
			ChatRoomClient.getInstance().receiveMessage(message);
		}
		return null;
	}
}
